### PR TITLE
feat(market): add sell history popup

### PR DESCRIPTION
**Pull request by Nyawmeowww**
### All Submissions

* [ ] Can you post a screenshot of your changes (if applicable)?

### MOBILE VIEW

**Buttons of "List NFTs" tab"**
image

**Tooltip for "Weapons Sold"**
image

**Tooltip for "Characters Sold"**
image

**Weapons Sold Modal**
image

**Characters Sold Modal**
image

**Weapons Sold Modal - No Sold Weapons**
image

**Characters Sold Modal - No Sold Characters**
image

**Weapons Sold Modal sorted by ID (Descending)**
image

**Characters Sold Modal sorted by ID (Ascending)**
image

### WEB VIEW

**Buttons of "List NFTs" tab"**
image

**Tooltip for "Weapons Sold"**
image

**Tooltip for "Characters Sold"**
image

**Weapons Sold Modal**
image

**Characters Sold Modal**
image

**Weapons Sold Modal - No Sold Weapons**
image

**Characters Sold Modal - No Sold Characters**
image

**Weapons Sold Modal sorted by ID (Descending)**
image

**Characters Sold Modal sorted by ID (Ascending)**
image

### New Feature Submissions

* [ ] Does this relate to an existing issue, or has it been talked about prior to being done (if a major change)? #293 

### Notes

Please note, your code must pass all tests and lint checks before it can be merged.

### PR Description

**Things done:**
* Responsive UI
* Design adjustments

_Notes:_ 
_* The link on the No Sold Weapon/s or Character/s Modal is this: https://cryptoblades.gitbook.io/wiki/market/marketplace#list-nfts_
_* The modal supports sorting by clicking the headers of the column they want to sort. See Characters Sold Modal sorted by ID (Ascending)_

### CODE REVIEW

File: frontend/src/interfaces/History.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+export interface IWeaponHistory{
+  buyerAddress: string
```
Comment (seiyria) [inline frontend/src/interfaces/History.ts:2]: all of these need semicolons. not sure why the linter isn't picking that up.

File: frontend/src/views/Market.vue [hunk 2]
```diff
@@ -1149,3 +1341,15 @@ export default Vue.extend({
 }
 
 </style>
+
+function getCharacterName(charId: string): String {
```
Comment (seiyria) [inline frontend/src/views/Market.vue:1345]: why are these here?

**Review by seiyria — CHANGES_REQUESTED** [review]

File: frontend/src/views/Market.vue [hunk 3]
```diff
@@ -436,13 +494,19 @@ interface Data {
...
-  weaponMarketTax: string ;  characterShowLimit: number;
+  weaponMarketTax: string ;
+  characterShowLimit: number;
   weaponShowLimit: number;
   allListingsAmount: number;
   currentPage: number;
   browseTabActive: boolean;
   listingSellPrice: string;
   priceChangeModal: boolean;
+  weaponTransactionHistoryData: any;
```
Comment (seiyria) [inline frontend/src/views/Market.vue:505]: why don't these use IWeaponHistory etc?

File: frontend/src/interfaces/History.ts (hunk 1 above)
Comment (Nyawmeowww) [reply frontend/src/interfaces/History.ts]: edited the semicolons

File: frontend/src/views/Market.vue (hunk 2 above)
Comment (Nyawmeowww) [reply frontend/src/views/Market.vue:1345]: Spaghetti codes. Removed line 1345 and the proceeding codes.

File: frontend/src/views/Market.vue (hunk 3 above)
Comment (Nyawmeowww) [reply frontend/src/views/Market.vue]: changed data types to their respective associations

**Review by seiyria — APPROVED** [review]
